python-checks/src/main/java/org/sonar/python/checks/LoopExecutingAtMostOnceCheck.java

```diff
@@ -77,7 +77,7 @@ private static void checkLoop(CfgBranchingBlock loopBlock, SubscriptionContext c
         return;
       }
       if (seen.add(b)) {
-        if (b.syntacticSuccessor() != null) {
+        if (b.syntacticSuccessor() != null && jumpOfCurrentLoop(b, loop)) {
```

Comment:
Because there is absolutely no documentation in the interface I have no idea what is the concept of a "syntacticSuccessor" (versus a normal successor), therefore I don't understand what is this condition doing.

Reply:
it's the block following this one if no jump is applied. It does apply to block having a jump statement as termination (i.e. break, continue, return, raise), otherwise it's null.
Good point, I will have documentation!